wiring/inc/spark_wiring_logging.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2016 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SPARK_WIRING_LOGGING_H
+#define SPARK_WIRING_LOGGING_H
+
+#include <initializer_list>
+#include <vector>
```

Comment:
You're right, another issue is heap fragmentation caused by `std::vector` when it grows. The problem is that I need to use some container that is able to grow dynamically during construction of the category mappings. I used `std::vector` mostly as a time-saver here, as all current loggers are intended to be instantiated globally before the application starts.